src/Crypto/RNCryptor/V3/Decrypt.hs

```diff
@@ -116,12 +128,13 @@ decrypt :: ByteString -> ByteString -> ByteString
 decrypt input pwd =
   let (rawHdr, rest) = B.splitAt 34 input
       -- remove the hmac at the end of the file
-      (toDecrypt, _) = B.splitAt (B.length rest - 32) rest
-      hdr = parseHeader rawHdr
+      (cipherText, msgHMAC) = B.splitAt (B.length rest - 32) rest
+      hdr = parseHeader $ rawHdr
       ctx = newRNCryptorContext pwd hdr
-      clearText = decryptCBC (ctxCipher ctx) (aesIV_ . rncIV . ctxHeader $ ctx) toDecrypt
-  in  removePaddingSymbols clearText
-
+      clearText = decrypt_ (ctxCipher ctx) (rncIV . ctxHeader $ ctx) cipherText
+      hmac = makeHMAC (rncHMACSalt . ctxHeader $ ctx) pwd $ rawHdr <> cipherText
+  in
+    if consistentTimeEqual msgHMAC hmac then removePaddingSymbols clearText else error "failed decrypt, invalid HMAC"
```

Comment:
Technically speaking, if my comment about equality holds, why not simply write `if msgHMAC == hmac ...` ?